plugins/optimization-detective/class-od-url-metric-group.php

```diff
@@ -239,7 +239,7 @@ public function is_complete(): bool {
 					$current_time > $url_metric->get_timestamp() + $this->freshness_ttl
 					||
 					// If the URL Metric's ETag doesn't match the current ETag, consider the URL metric as stale.
-					( null !== $this->collection && $url_metric->get_etag() !== $this->collection->get_current_etag() )
+					( null !== $this->collection && ! hash_equals( $url_metric->get_etag() ?? md5( '' ), $this->collection->get_current_etag() ) )
```

Comment:
I wonder if there is a better way to handle the absence of the ETag?

Reply:
How about https://github.com/WordPress/performance/pull/1705/files#r1864536539